app/utils/normalize-device-id.js

```diff
@@ -7,7 +7,21 @@ export function normalizeDeviceId(id) {
 
   if (length === 6) {
     if (RE_HEXDEC.test(id)) {
-      return `FLR${id}`;
+      // Perform guestimates based on 6-digit device ID.
+      // In case device ID is in the range D00000 to DFFFFF it is most likely a FLARM ID
+      if (parseInt(id, 16) >= 0xd00000 && parseInt(id, 16) <= 0xdfffff) {
+        return `FLR${id}`;
+      } else if (parseInt(id, 16) >= 0x004000 && parseInt(id, 16) <= 0xe94fff) {
+        // In case device ID is between 004000 to E94FFF it is most likely a ICAO-ID - this logic coule be elaborated more based on the country correlation
+        // See http://www.aerotransport.org/html/ICAO_hex_decode.html
+        return `ICA${id}`;
+      } else if (parseInt(id, 16) >= 0x000000 && parseInt(id, 16) <= 0x003fff) {
+        // In case device ID is between 0x000000 to 0x003FFF  assume it is an OGN ID
```

Comment:
what is the logic for this? are these common OGN IDs?

Reply:
This is the unallocated range of the ICAO HEX IDs taken from http://www.aerotransport.org/html/ICAO_hex_decode.html
OGN-IDs are derived from the CPU ID of the chipset the SW runs on and are therefore fully arbitrary AFAIK and don't respect any convention, unfortunately.

Reply:
IMHO we shouldn't default to OGN for anything unallocated. I would rather keep defaulting to Flarm or IGC in those cases.